src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,325 @@
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+
+    public function testCreateClientAllParameters(): void 
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $scriptId = $this->getScriptId($httpClient, 'script_all_true');
+        $clientJson = ClientMother::withAllParameters(
+            $clientName, 
+            1, 
+            "some description", 
+            false, 
+            2, 
+            [$scriptId], 
+            [$scriptId],
+            -1, 
+            "rsync long arguments", 
+            "rsync short arguments", 
+            "ssh arguments", 
+            "root@172.0.0.1"
+        );
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains($clientJson);
+        $this->assertHydraContext();
+    }
+    public function testCreateClientInvalidMaxParallelJobs(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::withMaxParallelJobs($clientName, -1);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(422);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            '@context' => '/api/contexts/ConstraintViolationList',
+            '@type' => 'ConstraintViolationList',
+            'hydra:title' => 'An error occurred',
+            'hydra:description' => 'maxParallelJobs: Max parallel jobs value must be a positive integer',
+        ]);
+    }
+
+    public function testCreateClientInvalidName(): void
+    {
+        $httpClient = $this->authenticate();
+        $clientName = $this->createClientName();
+        $clientJson = ClientMother::named($clientName);
+        $this->postClient($httpClient, $clientJson);
+        $this->postClient($httpClient, $clientJson);
+        $this->assertResponseStatusCodeSame(400);
+        $this->assertHydraError("An exception occurred while executing 'INSERT INTO Client (description, isActive, name, url, quota, sshArgs, rsyncShortArgs, rsyncLongArgs, state, maxParallelJobs, data, owner_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params [null, 1, \"".$clientName."\", \"\", -1, null, null, null, \"NOT READY\", 1, null, 1]:\n\nSQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '".$clientName."' for key 'Client.UNIQ_C0E801635E237E06'");
```

Comment:
Please open an issue so that eventually someone will fix and do a better error management, and put a comment here pointing to the recently opened issue

Reply:
I have opened the issue #554 to fix the problem